homeassistant/components/emoncms/config_flow.py

```diff
@@ -0,0 +1,170 @@
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import CONF_API_KEY, CONF_URL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import selector
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_MESSAGE,
+    CONF_ONLY_INCLUDE_FEEDID,
+    CONF_SUCCESS,
+    DOMAIN,
+    FEED_ID,
+    FEED_NAME,
+    FEED_TAG,
+    LOGGER,
+)
+
+
+def get_options(feeds: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    """Build the selector options with the feed list."""
+    return [
+        {
+            "value": feed[FEED_ID],
+            "label": f"{feed[FEED_ID]}|{feed[FEED_TAG]}|{feed[FEED_NAME]}",
+        }
+        for feed in feeds
+    ]
+
+
+def sensor_name(url: str) -> str:
+    """Return sensor name."""
+    sensorip = url.rsplit("//", maxsplit=1)[-1]
+    return f"emoncms@{sensorip}"
+
+
+async def get_feed_list(hass: HomeAssistant, url: str, api_key: str) -> dict[str, Any]:
+    """Check connection to emoncms and return feed list if successful."""
+    emoncms_client = EmoncmsClient(
+        url,
+        api_key,
+        session=async_get_clientsession(hass),
+    )
+    return await emoncms_client.async_request("/feed/list.json")
+
+
+class EmoncmsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """emoncms integration UI config flow."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlowWithConfigEntry:
+        """Get the options flow for this handler."""
+        return EmoncmsOptionsFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Initiate a flow via the UI."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {
+                    CONF_API_KEY: user_input[CONF_API_KEY],
+                    CONF_URL: user_input[CONF_URL],
+                }
+            )
+            result = await get_feed_list(
+                self.hass, user_input[CONF_URL], user_input[CONF_API_KEY]
+            )
+            if not result[CONF_SUCCESS]:
+                errors["base"] = result[CONF_MESSAGE]
+            else:
+                title = sensor_name(user_input[CONF_URL])
+                return self.async_create_entry(title=title, data=user_input)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_URL): str,
+                        vol.Required(CONF_API_KEY): str,
+                    }
+                ),
+                user_input,
+            ),
```

Comment:
We can now ask the user for what flows to import

Reply:
maybe I misunderstand something but just to be sure : you want I add something like a `async_step_choose_feeds` method in `EmoncmsConfigFlow` ? because I dont know if it is possible to use the `async_step_init` method of `EmoncmsOptionsFlow` in `EmoncmsConfigFlow`

If a new method is needed, we could keep this for a future PR don't you think ?

Reply:
I am just pointing this out because it was in the original design and now I discussed this with the core team we approve this. So if you still want to add it, go ahead.